### PR TITLE
Allow mods to prevent CA completion from counting towards campaign stats 

### LINKED ISSUE #752: Allow mods to prevent CA completion from counting towards campaign stats
Follow up to #696. Will need an allow/disallow event in `AnalyticsManager::OnCoverActionComplete`

### PR DESCRIPTION
**Pull request by Xymanek**
Closes #752

### CODE REVIEW

**Review by robojumper — APPROVED** [review]

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/AnalyticsManager.uc [hunk 1]
```diff
@@ -896,6 +896,13 @@ static function EventListenerReturn OnCoverActionComplete( Object EventData, Obj
 		return ELR_NoInterrupt;
 	}
 
+	// Start issue #752
+	if (!TriggerAllowOnCoverActionComplete(EventData, EventSource, GameState, Event, CallbackData))
```
Comment (pledbrook) [inline X2WOTCCommunityHighlander/Src/XComGame/Classes/AnalyticsManager.uc:900]: Shouldn't this be `TriggerAllowOnCovertActionComplete()`?

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/AnalyticsManager.uc [hunk 2]
```diff
@@ -908,6 +915,55 @@ static function EventListenerReturn OnCoverActionComplete( Object EventData, Obj
...
 
+// Start issue #752
+/// HL-Docs: feature:AllowOnCoverActionCompleteAnalytics; issue:752; tags:events,strategy
+/// Allows mods to prevent CA completion from counting towards campaign stats.
+///
+/// This event is triggered from `AnalyticsManager::OnCoverActionComplete` and passes all
+/// original listener arguments in the tuple.
+///
+/// ```unrealscript
+/// EventID: AllowOnCoverActionCompleteAnalytics
```
Comment (pledbrook) [inline X2WOTCCommunityHighlander/Src/XComGame/Classes/AnalyticsManager.uc:926]: Should be `AllowOnCovertActionCompleteAnalytics`

**Review by pledbrook — CHANGES_REQUESTED** [review]

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/AnalyticsManager.uc (hunk 1 above)
Comment (Xymanek) [reply X2WOTCCommunityHighlander/Src/XComGame/Classes/AnalyticsManager.uc:900]: image

Same response for the other comment

**Comment by pledbrook** [comment]
Someone else will have to merge this because I refuse to propagate a deliberate misspelling that will just cause mod authors grief when they can't work out why their event listeners aren't working.

**Review by robojumper — CHANGES_REQUESTED** [review]
I agree with Peter that this should use the correct spelling. Point out the discrepancy in the documentation if you'd like.

**Comment by Xymanek** [comment]
Rebased to resolve the merge conflict

**Review by pledbrook — APPROVED** [review]

**Comment by pledbrook** [comment]
@robojumper Do you want to re-review this, or shall we just move to "ready for merge" already?

**Review by robojumper — APPROVED** [review]

**Pull request merged**